itl/command-plugins.conf

```diff
@@ -549,6 +549,195 @@ object CheckCommand "http" {
 	vars.http_verbose = false
 }
 
+object CheckCommand "curl" {
+	import "ipv4-or-ipv6"
+
+	command = [ PluginDir + "/check_curl" ]
+
+	arguments += {
+		"-H" = {
+			value = "$curl_vhost$"
+			description = "Host name argument for servers using host headers (virtual host). Append a port to include it in the header (eg: example.com:5000)"
+		}
+		"-I" = {
+			value = "$curl_ip$"
+			description = "IP address or name (use numeric address if possible to bypass DNS lookup)."
+		}
+		"-p" = {
+			value = "$curl_port$"
+			description = "Port number (default: 80)"
+		}
+		"-4" = {
+			set_if = "$curl_ipv4$"
+			description = "Force `check_curl` to use IPv4 instead of choosing automatically"
+		}
+		"-6" = {
+			set_if = "$curl_ipv6$"
+			description = "Force `check_curl` to use IPv6 instead of choosing automatically"
+		}
+		"-S" = {
```

Comment:
While testing this, I experienced difficulties with the `-S` flag when an `-H` flag was not available as the Host object had no address attached directly.

Details: monitoring-plugins/monitoring-plugins#2013

> [PROBLEM] Service https _REDACTED_ on Not A Host is CRITICAL: HTTP CRITICAL - Unable to lookup IP address for '1.2+': getaddrinfo returned -4 - non-recoverable failure in name resolution